contracts/babylon/src/msg/contract.rs

```diff
@@ -26,36 +51,18 @@ pub struct InstantiateMsg {
     pub babylon_tag: String,
     pub btc_confirmation_depth: u32,
     pub checkpoint_finalization_timeout: u32,
-    /// notify_cosmos_zone indicates whether to send Cosmos zone messages notifying BTC-finalised
-    /// headers.
+    /// Whether to send Cosmos zone messages notifying BTC-finalised headers.
     /// NOTE: If set to true, then the Cosmos zone needs to integrate the corresponding message handler
     /// as well
     pub notify_cosmos_zone: bool,
-    /// If set, this will instantiate a BTC light client contract
-    pub btc_light_client_code_id: Option<u64>,
-    /// If set, this will define the instantiation message for the BTC light client contract.
-    /// This message is opaque to the Babylon contract, and depends on the specific light client
-    /// being instantiated
-    pub btc_light_client_msg: Option<Binary>,
-    /// If set, this will instantiate a BTC staking contract for BTC re-staking
-    pub btc_staking_code_id: Option<u64>,
-    /// If set, this will define the instantiation message for the BTC staking contract.
-    /// This message is opaque to the Babylon contract, and depends on the specific staking contract
-    /// being instantiated
-    pub btc_staking_msg: Option<Binary>,
-    /// If set, this will instantiate a BTC finality contract
-    pub btc_finality_code_id: Option<u64>,
-    /// If set, this will define the instantiation message for the BTC finality contract.
-    /// This message is opaque to the Babylon contract, and depends on the specific finality contract
-    /// being instantiated
-    pub btc_finality_msg: Option<Binary>,
-    /// If set, this will be the Wasm migration / upgrade admin of the BTC staking contract and the
-    /// BTC finality contract
+    /// Initialization info for the BTC light client contract.
+    pub btc_light_client_init_msg: Option<BtcLightClientInitMsg>,
+    /// Initialization info for the BTC staking contract.
+    pub btc_staking_init_msg: Option<BtcStakingInitMsg>,
+    /// Initialization info for the BTC finality contract.
+    pub btc_finality_init_msg: Option<BtcFinalityInitMsg>,
+    /// Wasm migration / upgrade admin of the BTC staking contract and the BTC finality contract.
```

Comment:
I recommend we avoid bundling code ID and instantiation message and keep these struct as is. They are separate things. Also, `InstantiateMsg` will have to be submitted as a JSON string via CLI, and CosmWasm provides poor support for nested JSON.

Reply:
I saw that the nested structure has been used in https://github.com/babylonlabs-io/cosmos-bsn-contracts/blob/49ea2a81de1b430afeb1951e4925ef9b4834529e/contracts/btc-finality/src/msg.rs#L15, so this is supposedly fine, I guess. Anyway, your other comment already renders this PR invalid, closing.